File: Examples/Objective-C/Remote Control/Remote Control/FlashlightViewController.m
//
//  Copyright © 2020 Bridgefy Inc. All rights reserved.
//

#import "FlashlightViewController.h"
#import "AdminViewTabBarController.h"

@interface FlashlightViewController ()

@property (weak, nonatomic) IBOutlet UIButton *flashlightButton;

@end

@implementation FlashlightViewController

- (IBAction)turnOnFlashlight:(id)sender {
    self.flashlightButton.enabled = NO;
    
    [self performSelector:@selector(enableFlashlightButton)
               withObject:nil
               afterDelay:15.0];
    
    if (self.tabBarController) {
        AdminViewTabBarController *tabBarController = (AdminViewTabBarController *)self.tabBarController;
        [tabBarController sendObject:nil sender:self];
    }
}

- (void)enableFlashlightButton {
    self.flashlightButton.enabled = YES;
}

@end

File: Examples/Swift/Remote Control/Remote Control/FlashlightViewController.swift
//
//  Copyright © 2020 Bridgefy Inc. All rights reserved.
//

import UIKit

class FlashlightViewController: UIViewController {
    
    @IBOutlet weak var flashlightButton: UIButton!

    @IBAction func turnOnFlashlight(_ sender: Any) {
        flashlightButton.isEnabled = false
        
        perform(#selector(enableFlashlightButton),
                     with: nil,
                     afterDelay: 15)
        
        if let tabBarController = tabBarController as? AdminViewTabBarController {
            tabBarController.send(object: nil, sender: self)
        }
    }
    
    @objc func enableFlashlightButton() {
        flashlightButton.isEnabled = true
    }

}
